packages/@react-aria/utils/src/useId.ts

```diff
@@ -15,17 +15,18 @@ import {useLayoutEffect, useMemo, useState} from 'react';
 let map: Map<string, (v: string) => void> = new Map();
 
 let id = 0;
-// don't want to conflict with ids from v2, this will guarantee something unique
+// don't want to conflict with ids from other instances of this module, this will *mostly* guarantee something unique
```

Comment:
so this will avoid collisions with different versions of the same module, but doesn't do anything for the case where two copies of the same version are loaded, which is something we might see in a micro frontend architecture
we'll need to still do something to protect against that
Unfortunately, that seems to indicate the need for a global :(
If you have any ideas, feel free to post them in thread here, I'll also confer with my team and see if they have more ideas

Reply:
> so this will avoid collisions with different versions of the same module, but doesn't do anything for the case where two copies of the same version are loaded, which is something we might see in a micro frontend architecture

exactly.  I ran into this just the other day when I was de-duping a dependency tree in one of our apps. This scenario can definitely happen, but it can also be solved (in npm with some fun dependency updating and/or `npm dedupe` usage, or in yarn with atlassian's `yarn-deduplicate` tool), but that is just a workaround for this issue that will certainly arise.

> Unfortunately, that seems to indicate the need for a global :(

I agree.  A WeakMap could be used to hold onto the versions for the different instances in play.

The question that remains is what should the expected behavior be?  It could:
- throw an error when a duplicate instance of same version is in use (force devs to dedupe)
- print a warning when a duplicate instance of same version is in use (nicer to dev workflow)
- allow multiple instances of the same version.  both instances share the same counter
- allow multiple instances of the same version, duplicate instances append -{n} to the id before the counter.

A lot of this seems like overkill.  the simplest seems to be:
- allow multiple instances of the same version.  both instances share the same counter

and at that point, probably don't even need to add the version number in there.  just a global counter for the id variable that all module instances can share from.  (this comment has me thinking this could all be that much simpler case:  https://github.com/facebook/react/issues/5867#issuecomment-354774705 .)